### PR TITLE
docs: added workflow hooks docs + changed workflow response

### PR DESCRIPTION
**Pull request by shahednasser**
- Add a documentation on workflow hooks. A lot of the details may later be removed to a documentation on how to hook into Medusa's core workflow hooks.
- Change the return value of workflows to be an instance of WorkflowResponse across the docs.

Closes DOCS-855

### CODE REVIEW

File: www/apps/book/app/advanced-development/workflows/add-workflow-hook/page.mdx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  title: `${pageNumber} Workflow Hooks`,
+}
+
+# {metadata.title}
+
+In this chapter, you'll learn what a workflow hook is, how to add it to your workflow, and how to add workflow hook handlers.
+
+## What is a Workflow Hook?
+
+A workflow hook is a point in a workflow where you can inject a step function, called a hook handler.
```
Comment (olivermrbl) [inline www/apps/book/app/advanced-development/workflows/add-workflow-hook/page.mdx]:
```suggestion
A workflow hook is a point in a workflow where you can inject custom functionality.
```

File: www/apps/book/app/advanced-development/workflows/add-workflow-hook/page.mdx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+Hook handlers receive an input from the workflow to perform an action using them. They are executed during the workflow's execution when it reaches the hook.
```
Comment (olivermrbl) [inline www/apps/book/app/advanced-development/workflows/add-workflow-hook/page.mdx]:
```suggestion
```

File: www/apps/book/app/advanced-development/workflows/add-workflow-hook/page.mdx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+- Your workflow isn't reusable by other application. Use a step that performs what a hook handler would instead.
+
+</Note>
+
+---
+
+## How to Add a Hook in a Workflow?
+
+To add a hook in your workflow, use the `createHook` function imported from `@medusajs/workflows-sdk`.
```
Comment (olivermrbl) [inline www/apps/book/app/advanced-development/workflows/add-workflow-hook/page.mdx]:
```suggestion
To expose a hook in your workflow, use the `createHook` function imported from `@medusajs/workflows-sdk`.
```

File: www/apps/book/app/advanced-development/workflows/add-workflow-hook/page.mdx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+The `createHook` function accepts two parameters:
+
+1. The first is a string indicating the hook's name. This is used to add a hook handler later.
+2. The second is the input to pass to the hook handler.
+
+The workflow must also return as a second parameter of `WorkflowResponse` an object having a `hooks` property. Its value is an array of the workflow's hooks.
+
+---
+
+## How to Add a Hook Handler?
```
Comment (olivermrbl) [inline www/apps/book/app/advanced-development/workflows/add-workflow-hook/page.mdx]:
```suggestion
## How to consume a Hook?
```

File: www/apps/book/app/advanced-development/workflows/add-workflow-hook/page.mdx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+Hook handlers are created in the `src/workflows/hooks` directory. They use the workflow's `hooks` property to add a hook handler.
```
Comment (olivermrbl) [inline www/apps/book/app/advanced-development/workflows/add-workflow-hook/page.mdx]:
```suggestion
Hook handlers are implemented by registering it on a given workflow in the `src/workflows/hooks` directory of your project. If a workflow exposes hooks, you should see the available ones on the `hooks` property of the workflow.
```

**Review by olivermrbl — APPROVED** [review]
LGTM, a few minor suggestion. Take them or leave them 💪

**Comment by shahednasser** [comment]
@olivermrbl thanks! I combined some of your suggestions with what was already there 👍🏻 We should leave merging until it's released.

**Pull request merged**